hutch_python/cli.py

```diff
@@ -98,14 +134,9 @@ def configure_ipython_session():
     # Set up tab completion modifications
     configure_tab_completion(ipy_config)
 
-    # remove force exit key bind from <ctrl-\\>
-    ipy_config.InteractiveShellApp.exec_lines = (
-        "import IPython; "
-        "from prompt_toolkit.keys import Keys; "
-        "ip = IPython.get_ipython(); "
-        "cb = Keys.ControlBackslash; "
-        "ip.pt_app.key_bindings.remove(cb) "
-    )
+    # Run startup hook code
+    lines = get_startup_hook(cfg=cfg)
+    ipy_config.InteractiveShellApp.exec_lines = lines
```

Comment:
I've managed to confuse myself reviewing this PR- what specific features are we trying to enable for the user via adding a post-init hook?

Reply:
My target has shifted to subscribing the elog callback specifically.  I thought giving users the ability to run something with their devices on hutch-python startup might be useful, but I'm also ok hiding that functionality if we think it crosses a line hutch-python shouldn't.

Reply:
I don't think it crosses a line per se I just failed to be imaginative enough to figure out what it would be used for. I think a lot of things can be done in `beamline.py`.

Reply:
Couldn't this be a minimal import-and-call like the following?

`import hutch_python; hutch_python.run_startup_hook()`

This gives the benefit of:
1. Lintable and checkable file as in this PR
2. Only 2 lines of source code that come without traceability and are extremely unlikely to fail (and if they do, we have bigger problems)
3. Much less stringified code compared to this PR